### ISSUE TITLE
Mail attempts to decode a  mail message twice when using exim

### ISSUE THREAD

**Issue by dukha** [mikel/mail#814]
When trying the basic example in the readme with exim, I only received and error:

> _NoMethodError: undefined method encoded' for #String:0x0000000947f9d8 from /home/mark/.rvm/gems/ruby-2.0.0-p247/gems/mail 2.5.4/lib/mail/network/delivery_methods/exim.rb:46:inblock in call'_

The mail message should have been an instance of the Mail class which respond to 'encoded', not a string.

It turns out the encoded is being called twice, once in the super class(Mail::Sendmail) and once in the subclass Mail::Exim.

My fix was to remove encoded from the Mail:Exim

```
def self.call(path, arguments, destinations, encoded_mail)
   popen "#{path} #{arguments}" do |io|
  io.puts encoded_mail.to_lf
  io.flush
   end
 end
```

This research was done in 2.5.4, however it looks like it is still in the master on github.

**Comment by bf4** [comment]
Thanks for this. Are you able to write a test that demonstrates this behavior?

**Comment by dukha** [comment]
Hi Ben
I could but I have time pressure right now. How much time to I have.

Regards

Mark

**Comment by cheenwe** [comment]
same question 

```sh
/root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/network/delivery_methods/exim.rb:47:in `block in call': undefined method `encoded' for #<String:0x007f8fd34eec28> (NoMethodError)
        from /root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/network/delivery_methods/sendmail.rb:73:in `popen'
        from /root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/network/delivery_methods/sendmail.rb:73:in `popen'
        from /root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/network/delivery_methods/exim.rb:46:in `call'
        from /root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/network/delivery_methods/sendmail.rb:57:in `deliver!'
        from /root/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/mail-2.6.4/lib/mail/message.rb:253:in `deliver!'
        from email.rb:17:in `<main>'
```

**Comment by jeremy** [comment]
Fixed by 67ac9bb7128c3e3f74ee4f8e7115a00727652876, backport of #689.
